src/main/scala/com/campudus/tableaux/cache/CacheClient.scala

```diff
@@ -0,0 +1,99 @@
+package com.campudus.tableaux.cache
+
+import com.campudus.tableaux.database.domain.DomainObject
+import com.campudus.tableaux.database.model.TableauxModel.{ColumnId, RowId, TableId}
+import io.vertx.core.eventbus._
+import io.vertx.core.{AsyncResult, Handler, Vertx}
+import io.vertx.scala.VertxExecutionContext
+import org.vertx.scala.core.json._
+
+import scala.concurrent.Future
+
+object CacheClient {
+  def apply(vertx: Vertx): CacheClient = {
+    new CacheClient(vertx: Vertx)
+  }
+}
+
+class CacheClient(override val _vertx: Vertx) extends VertxExecutionContext {
+
+  import io.vertx.scala.FunctionConverters._
+
+  type AsyncMessage[A] = Handler[AsyncResult[Message[A]]]
```

Comment:
I think you may be able to do `type AsyncJsonMessage = Handler[AsyncResult[Message[JsonObject]]]` here